File: three_files_urls/js/url_stuff.js
#!/usr/bin/env node

var h = [];
var stdin = process.openStdin();
stdin.setEncoding('utf8');

/* Process stdin */
var regexp = /embedCode=(\w+)/g;
stdin.on('data', function(line) {
  //process.stdout.write(line);
  while (match = regexp.exec(line)) { 
    h[match[1]] = 1; 
  }
});

/* When done, dump results */
process.stdin.on("end", function() {
  for (var k in h) { process.stdout.write(k + "\n"); }
});

File: three_files_urls/lua/url_stuff.lua
#!/usr/bin/env lua

local h = {}
for line in io.stdin:lines() do
  for m in string.gmatch(line, "embedCode=([%a%d]+)") do 
    h[m] = 1
  end
end

for k,v in pairs(h) do print(k) end
